src/domain-services/plans/plan-service.ts

```diff
@@ -57,25 +59,34 @@ export class PlanService {
     const planObjectsIDs: PlanId[] = plansFO.map((planFO) =>
       createBrandedValue(planFO.objectID)
     );
-    const plans: Array<InstanceDataOfModel<Database['plan']>> =
-      await models.plan.find({
+    const [plans, planVersionsByPlanId] = await Promise.all([
+      models.plan.find({
         where: {
           id: {
-            [Op.IN]: planObjectsIDs,
+            [models.Op.IN]: planObjectsIDs,
           },
         },
-      });
+      }),
+      groupObjectsByProperty(
+        await models.planVersion.find({
+          where: {
+            planId: {
+              [models.Op.IN]: planObjectsIDs,
+            },
+            currentVersion: true,
+          },
+        }),
+        'planId'
+      ),
+    ]);
 
     const plansMap = new Map<number, BasePlan[]>();
 
     for (const plan of plans) {
-      const planVersion = await models.planVersion.find({
-        where: {
-          planId: plan.id,
-          currentVersion: true,
-        },
-      });
-
+      const planVersion = planVersionsByPlanId.get(plan.id)?.at(0);
```

Comment:
If you listen to the advice above, you'd need a change here too
```suggestion
      const planVersion = planVersionsByPlanId.get(plan.id);
```